crates/script/src/simulate.rs

```diff
@@ -113,9 +113,10 @@ impl PreSimulationState {
         // Executes all transactions from the different forks concurrently.
         let futs = transactions
             .into_iter()
-            .map(|transaction| async {
+            .map(|mut transaction| async {
```

Comment:
why the mut?

Reply:
later it's used for `tx` with `transaction.tx_mut()`, but it's vestigial and not needed - had the same thing during my update

Reply:
The upstream foundry code https://github.com/foundry-rs/foundry/blob/736a3300234a0921b9d8adde6c0c4dd14053ec8a/crates/script/src/simulate.rs#L115-L150 doesn't make use of the mutability

Reply:
Yeah it was part of "submit upstream patches" on my side, just didn't get around much to it